### PR TITLE
DNSaaS User interface in Angular2

### PR DESCRIPTION
**Pull request by szok**
zrzut ekranu 2016-05-24 o 14 49 41
zrzut ekranu 2016-05-24 o 14 49 48
zrzut ekranu 2016-05-24 o 14 50 00
zrzut ekranu 2016-05-24 o 14 50 12
zrzut ekranu 2016-05-24 o 14 50 18
zrzut ekranu 2016-05-24 o 14 54 23
zrzut ekranu 2016-05-27 o 09 42 00
zrzut ekranu 2016-05-27 o 09 42 13

### CODE REVIEW

**Comment by mkurek** [comment]
any screenshots? :bowtie:

File: ui/static/app/auth/auth.service.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  public loggedIn: boolean = false;
+
+  constructor(
+      private http: Http,
+      private localStorage: LocalStorage
+  ) {
+    this.loggedIn = (this.localStorage.get("auth_token")) ? true : false;
+  }
+
+  login(username, password) {
```
Comment (ar4s) [inline ui/static/app/auth/auth.service.ts]: please add types

File: powerdns/views.py [hunk 2]
```diff
@@ -64,12 +64,19 @@ class DomainViewSet(OwnerViewSet):
     filter_fields = ('name', 'type')
     permission_classes = (DomainPermission,)
 
+    def get_queryset(self):
+        qs = super().get_queryset()
+        domain_name = self.request.query_params.get('domain_name')
```
Comment (ghost) [inline powerdns/views.py]: simple adding to `filter_fields` wont work?

Comment (szok) [inline powerdns/views.py]: Unfortunately, I need to look a icontains.
But maybe you have better idea?

Comment (mkurek) [inline powerdns/views.py]: you have `name` in `filter_fields` - it should work (using `name` query param, not `domain_name`).

Comment (mkurek) [inline powerdns/views.py]: consider http://django-filter.readthedocs.io/en/latest/usage.html#declaring-filters (together with http://www.django-rest-framework.org/api-guide/filtering/#specifying-a-filterset)

File: package.json [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    "test": "lite-server --open=ui/static/unit-tests.html",
+    "typings": "typings",
+    "postinstall": "typings install"
+  },
+  "license": "ISC",
+  "dependencies": {
+    "gulp": "3.9.1",
+    "gulp-typescript": "^2.8.0",
+    "gulp-sourcemaps": "1.6.0",
+    "gulp-tslint": "5.0.0",
```
Comment (mkurek) [inline package.json]: move it to dev dependencies

File: powerdns/serializers.py [hunk 4]
```diff
@@ -42,6 +43,7 @@ class Meta:
 class RecordSerializer(OwnerSerializer):
 
     id = ReadOnlyField()
+    domain_id = IntegerField(source='domain.id', read_only=True)
```
Comment (mkurek) [inline powerdns/serializers.py:59]: try to use `domain_id` here

File: ui/static/app/autocomplete/autocomplete.component.ts [hunk 5]
```diff
@@ -0,0 +1,141 @@
+import { Component, Input, AfterViewInit } from "@angular/core";
+import { AutocompleteServiceInterface } from "./autocomplete.service";
+
+
+@Component({
+  selector: "autocomplete",
+  template: `
```
Comment (mkurek) [inline ui/static/app/autocomplete/autocomplete.component.ts]: is it possible to read this template from a file?

File: ui/templates/ui/index.html [hunk 6]
```diff
@@ -0,0 +1,21 @@
+<html>
+<head>
+  <title>DNSaaS</title>
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css">
```
Comment (mkurek) [inline ui/templates/ui/index.html]: download it and don't use cdn here

**Comment by ghost** [comment]
:ok:

**Comment by dominikkowalski** [comment]
:ok:

**Comment by mkurek** [comment]
:ok:

**Comment by ar4s** [comment]
:ok:

**Pull request merged**